### PR TITLE
[chore] Circle ci update

### PR DESCRIPTION
**Pull request by stackchain**
- Move from old legacy circleci repo to cimg 
- Bump node img version to 16.5.0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,7 +93,7 @@
     "react-native": "0.63.4",
     "react-native-background-timer": "2.4.1",
     "react-native-ble-plx": "2.0.2",
-    "react-native-blockies-svg": "Emurgo/react-native-blockies-svg",
+    "react-native-blockies-svg": "git+https://github.com/Emurgo/react-native-blockies-svg",
```
Comment (wolverineks) [inline package.json:96]: what does this do?

File: .circleci/config.yml [hunk 2]
```diff
@@ -37,43 +32,33 @@ jobs:
...
   test:
     <<: *rn_defaults
     steps:
       - attach_workspace:
           at: ~/project
-      - run:
-          name: Match bitrise's yarn version
-          command: sudo npm install -g yarn@${BITRISE_YARN_VER}
-      - run: yarn test
+      - run: yarn test --runInBand
```
Comment (wolverineks) [inline .circleci/config.yml:52]: why run serially, instead of parallel?

**Comment by wolverineks** [comment]
does circleci add anything beyond GitHub actions? could we ditch this altogether? 

if Android job isn't running, I'd like to get rid of the job

if Android job isn't running, do we still need to install rust?

File: package.json (hunk 1 above)
Comment (stackchain) [reply package.json:96]: When adding modules that use git protocol over yarn/npm adding this way makes it compatible with both.

File: .circleci/config.yml (hunk 2 above)
Comment (stackchain) [reply .circleci/config.yml:52]: We are using a free tier in circle-ci, it was the only way to make it run, we were getting too often out-of-memory when running in parallel.

File: package.json (hunk 1 above)
Comment (wolverineks) [reply package.json:96]: why is this necessary?

Comment (stackchain) [reply package.json:96]: The project must be agnostic about the package tool, before it was tight to yarn.

**Comment by stackchain** [comment]
> does circleci add anything beyond GitHub actions? could we ditch this altogether?
> 
> if Android job isn't running, I'd like to get rid of the job
> 
> if Android job isn't running, do we still need to install rust?

@vsubhuman has a plan to remove the ci pipeline from dev we will keep it only for releases probably, but that is not covered in this PR.
There is no longer the android job in ci.

**Pull request merged**